### PR TITLE
Diona feel no pain

### PR DESCRIPTION
**Pull request by Eimras**
Adds a check for Diona on pain(I think), and a check to electrocution to prevent Diona from being electrocuted. May they work on wiring and airlock repairs again!

### CODE REVIEW

**Comment by Anewbe** [comment]
... Also makes them immune to certain technomancer spells, which may or may not be ☹️ . Will need to discuss with staff.

**Comment by Neerti** [comment]
Might be better to replace the hard Diona checks with ```species.flags & NO_PAIN``` instead of ```species.get_bodytype() == "Diona"```.

**Comment by Eimras** [comment]
Scoped down the shocking immunity to only the hands of species with the IS_PLANT flag and changed the pain checking to check NO_PAIN flag.

**Comment by Anewbe** [comment]
You test the zapping part?

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -966,9 +966,11 @@ default behaviour is:
 /mob/living/proc/update_water() // Involves overlays for humans.  Maybe we'll get submerged sprites for borgs in the future?
 	return
 
-/mob/living/proc/can_feel_pain(var/check_organ)
+/mob/living/proc/can_feel_pain(var/check_organ, var/mob/living/carbon/human/H)
 	if(isSynthetic())
 		return FALSE
+	if(H.species.flags & NO_PAIN)
```
Comment (Neerti) [inline code/modules/mob/living/living.dm]: Think it'd be better to override the human's ```can_feel_pain()``` and check for species instead.

**Comment by Eimras** [comment]
The zapping part was tested and works as intended. The can_feel_pain proc is also in human, and that one does already check for NO_PAIN so my change to living is a bit redundant.

**Pull request closed without merging**